typescript/token/contracts/libs/FastTokenRouter.sol

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {Message} from "./Message.sol";
+import {TokenRouter} from "./TokenRouter.sol";
+
+import {TypeCasts} from "@hyperlane-xyz/core/contracts/libs/TypeCasts.sol";
+
+/**
+ * @title Common FastTokenRouter functionality for ERC20 Tokens with remote transfer support.
+ * @author Abacus Works
+ */
+abstract contract FastTokenRouter is TokenRouter {
+    using TypeCasts for bytes32;
+    using Message for bytes;
+    /**
+     * @notice `FastTransferMetadata` is the LP data stored against `fastTransferId`.
+     */
+    struct FastTranferMetadata {
+        address filler;
+        address recipient;
+        uint256 amount;
+        uint256 fastFee;
+    }
+
+    uint256 public fastTransferId;
+    // maps `fastTransferId` to `FastTranferMetadata`.
+    mapping(bytes32 => FastTranferMetadata) filledFastTransfers;
```

Comment:
instead of storing the whole struct can you store a hash and only use a single slot?